### PR TITLE
Add `Þe` element

### PR DESCRIPTION
**Pull request by Lyxal**
This is what you meant right?

Goes towards closing #807

### CODE REVIEW

**Comment by Lyxal** [comment]
Dang. Could someone fix the test case that's erroring for me?

**Comment by Lyxal** [comment]
It's got too many brackets which is probably a typo I missed because the checks didn't run after the continuous integration commit. And now it's too late time wise for me to fix rn

**Review by chunkybanana — APPROVED** [review]

File: vyxal/elements.py [hunk 1]
```diff
@@ -2379,6 +2379,24 @@ def matrix_determinant(lhs, ctx):
...
 
+def matrix_exponentiation(lhs, rhs, ctx):
+    """Element Þe
+    (lst, num) -> (a * a) b times
+    (num, lst) -> (b * b) a times
+    """
+
+    ts = vy_type(lhs, rhs, simple=True)
+    if set(ts) != {list, NUMBER_TYPE}:
+        raise TypeError("matrix exponentiation only works on lists and numbers")
```
Comment (cgccuser) [inline vyxal/elements.py:2390]: Maybe say instead "Matrix exponentiation requires a matrix and a number"? This implies that two numbers or two matrices would work

File: vyxal/elements.py [hunk 2]
```diff
@@ -2379,6 +2379,24 @@ def matrix_determinant(lhs, ctx):
...
+
+    matrix, times = lhs, rhs if ts[0] == NUMBER_TYPE else rhs
+    original_matrix = deep_copy(matrix)
```
Comment (cgccuser) [inline vyxal/elements.py:2393]: There's no need to copy, is there? `multiply` doesn't mutate its arguments

**Review by cgccuser — CHANGES_REQUESTED** [review]
LGTM, just a couple minor things

File: vyxal/elements.py (hunk 2 above)
Comment (Lyxal) [reply vyxal/elements.py:2393]: Sure, it doesn't. But assignment does.

**Review by PyGamer0 — APPROVED** [review]

File: vyxal/elements.py (hunk 2 above)
Comment (cgccuser) [reply vyxal/elements.py:2393]: @Lyxal It doesn't, `original_matrix = matrix` is enough

File: vyxal/elements.py [hunk 3]
```diff
@@ -2379,6 +2379,24 @@ def matrix_determinant(lhs, ctx):
...
+    (lst, num) -> (a * a) b times
+    (num, lst) -> (b * b) a times
+    """
+
+    ts = vy_type(lhs, rhs, simple=True)
+    if set(ts) != {list, NUMBER_TYPE}:
+        raise TypeError("Matrix exponentiation requires a matrix and a number")
+
+    matrix, times = lhs, rhs if ts[0] == NUMBER_TYPE else rhs
+    original_matrix = deep_copy(matrix)
```
Comment (cgccuser) [inline vyxal/elements.py:2393]:
```suggestion
    original_matrix = matrix
```

**Review by cgccuser — CHANGES_REQUESTED** [review]

File: vyxal/elements.py (hunk 3 above)
Comment (cgccuser) [reply vyxal/elements.py:2393]: If this doesn't work, something is wrong with the rest of our code

**Review by cgccuser — APPROVED** [review]
LGTM

**Pull request merged**